eclair-core/src/main/scala/fr/acinq/eclair/router/Graph.scala

```diff
@@ -90,7 +90,9 @@ object Graph {
             val neighbor = edge.desc.b
 
             // note: the default value here will never be used, as there is always an entry for the current in the 'cost' map
-            val newMinimumKnownCost = cost.get(current.key) + edgeWeightByAmount(edge, amountMsat)
+            // note: when searching on a reversed graph we consider 0 the weight of the edge of the target-neighboring node
+            // this is because it will be a direct channel to the source and it pays no fees
+            val newMinimumKnownCost = cost.get(current.key) + edgeWeightByAmount(edge, amountMsat, neighbor == targetNode)
```

Comment:
I don't think this is right: the fees that are applied are the fees for the outgoing edge, and which is then added to the amount that we want to forward. Suppose we have:

A
    \
     C -- D
  /
B
And want to forward X from A to D.
First we look for incoming edges of D that can carry X. We have one: edge_cd
Then we move backwards and look for incoming edges of C that can carry amount + cost(edge_cd, amount) and so on: the amount that is used to compute the relay fee is updated at each hop.